lib/board.rb

```diff
@@ -0,0 +1,67 @@
+
+class Board
+  attr_accessor :cells
+
+  def initialize
+    self.reset!
+
+  end
+
+  def reset!
+    self.cells = [" "] * 9
+  end
+
+  def display
+    d = self.cells
+
+    puts "-----------"
+    puts " #{d[0]} | #{d[1]} | #{d[2]} "
+    puts "-----------"
+    puts " #{d[3]} | #{d[4]} | #{d[5]} "
+    puts "-----------"
+    puts " #{d[6]} | #{d[7]} | #{d[8]} "
+    puts "-----------"
+  end
+
+  def position(location)
+    c = self.cells
+
+    c[location.to_i - 1]
+  end
+
+  def turn_count
+    c = self.cells
+
+    c.select{|e| e.strip != ""}.size
+  end
+
+  def full?
+    count = self.turn_count
```

Comment:
We can also get this value without calling #turn_count by using the [`#all?`](http://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-all-3F) iterator.

`cells.all? {|cell| cell != " "}`